src/Dapr.Client/DaprClient.cs

```diff
@@ -714,6 +714,17 @@ public HttpRequestMessage CreateInvokeMethodRequest<TRequest>(string appId, stri
         /// <returns>A <see cref="Task{IReadOnlyList}" /> that will return the list of values when the operation has completed.</returns>
         public abstract Task<IReadOnlyList<BulkStateItem>> GetBulkStateAsync(string storeName, IReadOnlyList<string> keys, int? parallelism, IReadOnlyDictionary<string, string> metadata = default, CancellationToken cancellationToken = default);
 
+        /// <summary>
+        /// Gets a list of deserialized values associated with the <paramref name="keys" /> from the Dapr state store.
```

Comment:
I might add some remarks to be clear about the assumption that all keys in the bulk operation are expected to be of the specified type (and maybe refer to the original method for cases where mixed types might be retrieved).

Reply:
Added more context to the comment per your suggestion.